### PR TITLE
feat(presets): add 'SPFx' monorepo

### PR DESCRIPTION
**Pull request by fzoske**
## Changes

Add spfx monorepo

https://aka.ms/spfx

## Context

All packages need to be in the same version (https://learn.microsoft.com/en-us/sharepoint/dev/spfx/release-1.15.2)

Screenshot 2022-10-05 at 12 24 23

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by viceice — CHANGES_REQUESTED** [review]
why not using grouping via sourceUrl, like the other monorepos above?

**Comment by fzoske** [comment]
I could not find the source for the packages, aka.ms/spfx is a documentation page
On the other hand @microsoft/eslint-plugin-spfx for exmample does not have an associated url: https://www.npmjs.com/package/@microsoft/eslint-plugin-spfx

**Comment by JamieMagee** [comment]
I can confirm that these packages are not released on GitHub

**Review by JamieMagee — CHANGES_REQUESTED** [review]
Requires lint fixes

**Review by rarkins — DISMISSED** [review]

**Review by JamieMagee — DISMISSED** [review]

File: lib/config/presets/internal/monorepo.ts [hunk 1]
```diff
@@ -246,6 +246,7 @@ const patternGroups = {
   clarity: ['^@cds/', '^@clr/'],
   embroider: '^@embroider/',
   fullcalendar: '^@fullcalendar/',
+  spfx: ['^@microsoft/sp-', '^@microsoft/eslint-.*-spfx'],
```
Comment (viceice) [inline lib/config/presets/internal/monorepo.ts:249]: can it match any false positive package?
```suggestion
  spfx: ['^@microsoft\\/sp-', '^@microsoft\\/eslint-.+-spfx$'],
```

**Review by rarkins — APPROVED** [review]

**Review by viceice — APPROVED** [review]
hopefully we don't get false positives 🤷‍♂️

**Pull request merged**